### PR TITLE
feat: Offer type annotations for exposed functions.

### PR DESCRIPTION
**Pull request by isidentical**
image
image
image
image

### CODE REVIEW

File: src/fal/context.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+# don't manually alter the contents.
+
+from __future__ import annotations
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    import pandas as pd
+    from faldbt.project import DbtModel, DbtTest, Feature
+    from fal.fal_script import Context
+    from typing import Optional, Dict, List, Protocol, Any
```
Comment (isidentical) [inline src/fal/context.py:11]: The reason we use protocols here is the `Callable[]` syntax can't really represent some stuff (like defaults and doc-strings) https://mypy.readthedocs.io/en/stable/protocols.html#callback-protocols

**Review by chamini2 — APPROVED** [review]
This is PRETTY COOL!

File: integration_tests/projects/008_pure_python_models/scripts/post_hook.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 from functools import reduce
 import os
 import time
+from fal.context import *
```
Comment (turbo1912) [inline integration_tests/projects/008_pure_python_models/scripts/post_hook.py:5]: and everything still works if you dont import this right?

**Review by turbo1912 — APPROVED** [review]

File: integration_tests/projects/008_pure_python_models/scripts/post_hook.py (hunk 2 above)
Comment (isidentical) [reply integration_tests/projects/008_pure_python_models/scripts/post_hook.py:5]: yep!

**Pull request merged**